plugin/parser/differ/mysql/differ.go

```diff
@@ -139,33 +145,95 @@ func (*SchemaDiffer) SchemaDiff(oldStmt, newStmt string) (string, error) {
 			}
 
 			if len(alterTableDropConstraintSpecs) > 0 {
-				diff = append(diff, &ast.AlterTableStmt{
-					Table: &ast.TableName{
-						Name: model.NewCIStr(tableName),
+				drop = append(drop, []ast.Node{
+					&ast.AlterTableStmt{
+						Table: &ast.TableName{
+							Name: model.NewCIStr(tableName),
+						},
+						Specs: alterTableAddConstraintSpecs,
 					},
-					Specs: alterTableAddConstraintSpecs,
 				})
 			}
 		default:
 		}
 	}
 
-	return deparse(diff, format.DefaultRestoreFlags|format.RestoreStringWithoutCharset)
+	return deparse(add, inplaceUpdate, nonInplaceDrop, nonInplaceAdd, drop, format.DefaultRestoreFlags|format.RestoreStringWithoutCharset)
 }
 
-func deparse(nodes []ast.Node, flag format.RestoreFlags) (string, error) {
+func deparse(add []ast.Node, inplaceUpdate []ast.Node, nonInplaceDrop [][]ast.Node, nonInplaceAdd []ast.Node, drop [][]ast.Node, flag format.RestoreFlags) (string, error) {
 	var buf bytes.Buffer
-	for _, node := range nodes {
+	// We should following the right order to avoid break the dependency:
+	// Additions for new nodes.
+	// Updates for in-place node updates.
+	// Deletions for destructive (none in-place) node updates (in reverse order).
+	// Additions for destructive node updates.
+	// Deletions for deleted nodes (in reverse order).
+	for _, node := range add {
```

Comment:
According to this comment, I think we have more readable names for these variables.
How about rename `add` to `newNodeList`? 
And what is the difference between Updates and Additions for destructive?

Reply:
> And what is the difference between Updates and Additions for destructive?

Like [this](https://github.com/bytebase/bytebase/pull/2971#discussion_r993057581).